linkerd/stack/src/watch.rs

```diff
@@ -48,31 +47,50 @@ pub struct SpawnWatch<S> {
 // === impl NewSpawnWatch ===
 
 impl<P, N> NewSpawnWatch<P, N> {
-    pub fn new(new_update: N) -> Self {
+    pub fn new(inner: N) -> Self {
         Self {
-            new_update,
+            inner,
             _marker: std::marker::PhantomData,
         }
     }
+
+    pub fn layer() -> impl tower::layer::Layer<N, Service = Self> + Clone {
+        crate::layer::mk(Self::new)
+    }
+
+    // Creates a new `NewSpawnWatch` layer that transforms the target type to
+    // `T` for the inner stack.
+    pub fn layer_into<T>(
```

Comment:
naming nit, not a big deal: i might prefer to call this `layer_from_tuple`, or something, since the inner `NewService` is `NewWatchedFromTuple` rather than `NewWatchedInto`...

Reply:
yeah, but into is much clearer at the callsite, in my opinion. The caller doesn't care which middleware is used to achieve the into.

Reply:
I suppose _from_targets_ could work. The _tuple_ is kind of a non-sequitur from the caller's point of view, though.

Reply:
fair enough, i'm fine with whatever you think is clearer :+1: